packages/p2p/src/components/app.jsx

```diff
@@ -1,66 +1,46 @@
 import classNames from 'classnames';
 import React from 'react';
+import { observer } from 'mobx-react-lite';
 import PropTypes from 'prop-types';
-import { getPropertyValue, isProduction } from '@deriv/shared';
+import { getPropertyValue } from '@deriv/shared';
 import { Tabs, Modal } from '@deriv/components';
 import { Dp2pProvider } from 'Components/context/dp2p-context';
 import ServerTime from 'Utils/server-time';
-import { init as WebsocketInit, getModifiedP2POrderList, requestWS, subscribeWS, waitWS } from 'Utils/websocket';
+import { subscribeWS, waitWS } from 'Utils/websocket';
 import { localize, setLanguage } from './i18next';
-import OrderInfo, { orderToggleIndex } from './orders/order-info';
 import BuySell from './buy-sell/buy-sell.jsx';
 import MyAds from './my-ads/my-ads.jsx';
 import Orders from './orders/orders.jsx';
 import NicknameForm from './nickname/nickname-form.jsx';
 import Download from './verification/download.jsx';
 import Verification from './verification/verification.jsx';
+import { useStores } from '../../stores';
 import './app.scss';
 
 const allowed_currency = 'USD';
-
-const path = {
-    buy_sell: 0,
-    orders: 1,
-    my_ads: 2,
-    // my_profile: 3,
-};
-
-class App extends React.Component {
-    is_mounted = false;
-
-    constructor(props) {
-        super(props);
-
-        setLanguage(this.props.lang);
-        WebsocketInit(this.props.websocket_api, this.props.client.local_currency_config.decimal_places);
-        ServerTime.init(this.props.server_time);
-
-        this.ws_subscriptions = {};
-        this.list_item_limit = 20;
-        this.is_active_tab = true;
-        this.state = {
-            active_index: 0,
-            loginid: this.props.client.loginid,
-            order_offset: 0,
-            orders: [],
-            notification_count: 0,
-            active_notification_count: 0,
-            inactive_notification_count: 0,
-            parameters: null,
-            is_advertiser: false,
-            is_restricted: false,
-            show_popup: false,
-            order_table_type: orderToggleIndex.ACTIVE,
-            chat_info: {
-                app_id: '',
-                user_id: '',
-                token: '',
-            },
-        };
-    }
-
-    componentDidMount() {
-        this.is_mounted = true;
+const App = observer(props => {
+    const { general_store } = useStores();
+    const {
+        className,
+        custom_strings,
+        is_mobile,
+        lang,
+        order_id,
+        poi_url,
+        server_time,
+        setOrderId,
+        should_show_verification,
+        websocket_api,
+    } = props;
+    const is_mounted = React.useRef(false);
```

Comment:
We can get rid of this `is_mounted`.

We're currently only using this in the componentDidMount hook and since we're using Mobx now we won't need to worry about async functions trying to update a component's state, Mobx will handle this. 👍

Reply:
Alright, this is done. Thank you!